ext/drx/drx.h

```diff
@@ -459,6 +459,49 @@ void
 drx_buf_insert_buf_memcpy(void *drcontext, drx_buf_t *buf, instrlist_t *ilist,
                           instr_t *where, reg_id_t dst, reg_id_t src, ushort len);
 
+DR_EXPORT
+/**
+ * Expands AVX2 gather and AVX-512 gather and scatter instructions to a sequence of
+ * equivalent scalar load and stores, mask register bit tests, and mask register bit
+ * updates.
+ *
+ * WARNING: This function is not fully supported yet. Do not use.
+ *
+ * WARNING: The added multi-instruction loop contains several control-transfer
+ * instructions and is not straight-line code, which can complicate subsequent analysis
+ * routines.
+ *
+ * WARNING: This function makes a call to drreg_init() and drreg_exit(), extending the
```

Comment:
It seems like a waste to call init and exit on every single bb!  Lazy init seems fine, but why do repeated alloc..free 100,000 times?

Reply:
I can make it a one-time lazy. globals in extensions are not protected are they? (I don't see where they would be)